sqlmesh/core/model/definition.py

```diff
@@ -634,7 +633,10 @@ def depends_on_past(self) -> bool:
             query = self.render_query(optimize=False)
             if query is None:
                 return False
-            return self.name in d.find_tables(query, dialect=self.dialect)
+            tables = d.find_tables(
+                query, default_catalog=self.default_catalog, dialect=self.dialect
+            )
+            return self.name in tables or self.fqn in tables
```

Comment:
why do you need to check name and self.fqn? it should just be self.fqn because find_tables is normalized